trade-finance-manager-ui/server/controllers/utilisation-reports/find-reports-by-year/search-input-validator.ts

```diff
@@ -0,0 +1,60 @@
+import { isNonEmptyString } from '@ukef/dtfs2-common';
+import { REGEX } from '../../../constants';
+
+type Error = {
+  text: string;
+  href: string;
+};
+
+export type FindUtilisationReportByYearValidationErrors = {
+  errorSummary: Error[];
+  bankError: string | undefined;
+  yearError: string | undefined;
+};
+
+const isBefore2024 = (year: number): boolean => year < 2024;
+
+const isInTheFuture = (year: number): boolean => year > new Date().getFullYear();
+
+const validateYearInputAndReturnErrorMessageIfInvalid = (year: unknown): string | undefined => {
+  const validYearRegex = new RegExp(REGEX.YEAR);
+  if (!year || !isNonEmptyString(year) || !validYearRegex.test(year)) {
+    return 'Enter a valid year';
+  }
+  const yearParsed = Number(year);
+  if (isBefore2024(yearParsed) || isInTheFuture(yearParsed)) {
+    return 'Enter a year that is between 2024 and the current year';
+  }
+  return undefined;
+};
+
+const validateBankInputAndReturnErrorMessageIfInvalid = (bankId: unknown, allBankIds: string[]): string | undefined => {
+  if (!bankId || !isNonEmptyString(bankId) || !allBankIds.includes(bankId)) {
+    return 'Select a bank';
+  }
+  return undefined;
+};
+
+export const validateSearchInput = (
+  bankId: unknown,
+  year: unknown,
+  allBankIds: string[],
```

Comment:
I think these parameters should be wrapped up in an object rather than being defined one after another in this case